caravel/assets/javascripts/SqlLab/components/QuerySearch.jsx

```diff
@@ -34,13 +36,41 @@ class QuerySearch extends React.Component {
     const val = (db) ? db.value : null;
     this.setState({ databaseId: val });
   }
-  insertParams(baseUrl, params) {
-    return baseUrl + '?' + params.join('&');
+  getTimeFromSelection(selection) {
+    switch (selection) {
```

Comment:
I tried using the dict from value to function, but the function is evaluated when the dict is defined. There might be a way to delay the evaluation, but I figured switch statements might be a better option for mapping to functions?